harness/tests/determined/pytorch/experimental/test_torch_batch_process.py

```diff
@@ -0,0 +1,570 @@
+import math
+from typing import Any, Dict, List, Optional, Type
+from unittest.mock import MagicMock, Mock, call, patch
+
+import pytest
+import torch
+from torch.utils.data import Dataset
+
+from determined import pytorch
+from determined.pytorch.experimental._torch_batch_process import (
+    TorchBatchProcessor,
+    TorchBatchProcessorContext,
+    get_default_device,
+    torch_batch_process,
+)
+
+DEFAULT_SLOT_IDS = [0]
+DEFAULT_ADDRS = ["0.0.0.12"]
+DEFAULT_DEVICE = torch.device("cuda", 1)
+DEFAULT_STORAGE_PATH = "default_storage_path"
+
+
+@pytest.fixture
+def batch_processor_with_avg_metric_reducer() -> Type[TorchBatchProcessor]:
```

Comment:
I don't think this needs to be a fixture.  I think you could just define MySumMetricReducer and MyProcessor directly in the file and reference MyProcessor normally.

Fixtures are really for resources that need to be stood up for a test and torn down after a test, like creating a temporary directory and deleting it when the test finishes.